ndc-client/src/apis/mod.rs

```diff
@@ -4,15 +4,19 @@ use std::fmt;
 #[derive(Debug, Clone)]
 pub struct ConnectorError {
     pub status: reqwest::StatusCode,
-    pub error_response: crate::models::ErrorResponse,
+    pub error_response: Option<crate::models::ErrorResponse>,
```

Comment:
The spec does require that the connector return an `ErrorResponse`, so I think we should call this case out as a new top-level constructor, so that it's more obvious on the engine side what ought to be done with it (we should capture the original status code, but report this as a bug on the connector ideally).

Reply:
This would mean converting `ErrorResponse` to an `enum`. Are you OK with that?

Reply:
I still think the key is that the caller needs to be able to identify this special case and report it (without having to case on strings).  I don't think `ErrorResponse` needs to be an enum, you could just add a new alternative to `ndc_client::Error`, called `InvalidErrorResponse` or something similar, which captures the useful parts of the response object (or maybe all of it).

Reply:
Good idea. I pushed a change.